scripts/bootstrap.sh.in

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+set -e
+set -u
+if test -z "$(type -p)"; then set -o pipefail; fi
+
+RPM=${RPM:-$(command -v rpm)}
+SYSTEMCTL=${SYSTEMCTL:-$(command -v systemctl)}
+YUM=${YUM:-$(command -v yum)}
+SALT_CALL=${SALT_CALL:-salt-call}
+
+SALT_MINION_FILE_CLIENT_LOCAL_CONF=/etc/salt/minion.d/99-file-client-local.conf
+
+die() {
+        >/dev/stderr echo "$@"
+        exit 1
+}
+
+pre_minion_checks() {
+        test "x$(whoami)" = "xroot" || die "Script must run as root"
+        test -n "${RPM}" || die "rpm not found"
+        test -x "${RPM}" || die "rpm at '${RPM}' is not executable"
+        test -n "${SYSTEMCTL}" || die "systemctl not found"
+        test -x "${SYSTEMCTL}" || die "systemctl at '${SYSTEMCTL}' is not executable"
+        test -n "${YUM}" || die "yum not found"
+        test -x "${YUM}" || die "yum at '${YUM}' is not executable"
+}
+
+disable_salt_minion_service() {
+        ${SYSTEMCTL} disable salt-minion.service 2>/dev/null || true
+}
+
+stop_salt_minion_service() {
+        ${SYSTEMCTL} stop salt-minion.service 2>/dev/null || true
+}
+
+configure_salt_repository() {
+        ${RPM} --import https://repo.saltstack.com/yum/redhat/7/x86_64/archive/2018.3.3/SALTSTACK-GPG-KEY.pub
+        cat > /etc/yum.repos.d/saltstack.repo << EOF
+[saltstack-repo]
+name=SaltStack repo for RHEL/CentOS \$releasever
+baseurl=https://repo.saltstack.com/yum/redhat/\$releasever/\$basearch/archive/2018.3.3
+enabled=1
+gpgcheck=1
+gpgkey=https://repo.saltstack.com/yum/redhat/\$releasever/\$basearch/archive/2018.3.3/SALTSTACK-GPG-KEY.pub
+EOF
+        ${YUM} clean expire-cache
+}
+
+install_salt_minion() {
+        ${YUM} install -y salt-minion
+}
+
+configure_salt_minion_local_mode() {
+        cat > "${SALT_MINION_FILE_CLIENT_LOCAL_CONF}" << EOF
+file_client: local
+file_roots:
+  metalk8s-@VERSION@:
+    - /srv/scality/metalk8s-dev/salt
+pillar_roots:
+  metalk8s-@VERSION@:
+    - /srv/scality/metalk8s-dev/pillar
+EOF
+}
+
+run_bootstrap_prechecks() {
+        return
+}
+
+install_kubelet() {
+        ${SALT_CALL} --local --retcode-passthrough state.apply metalk8s.containerd saltenv=metalk8s-@VERSION@ pillarenv=metalk8s-@VERSION@
```

Comment:
You don't need --local here as you put `file_client: local` in salt minion config

You can also put pillarenv directly in the minion config or maybe we should set `pillarenv_from_saltenv: True` in the config

Reply:
I know this is not strictly required. However, I like to keep it around: the environment in which this execution takes place is 'a bit odd', and being very explicit about this (e.g. passing in `--local`) makes this more apparent.